plugins/locale.js

```diff
@@ -1,24 +1,54 @@
-import { join } from "node:path";
 import { existsSync, readFileSync } from "node:fs";
 import fp from "fastify-plugin";
 import chalk from "chalk";
 import emojiFlags from "emoji-flags";
 
-// function that takes a locale string and returns the emoji flag for that locale
-// handles formats like en-US, en, en-US-POSIX
-// map countries to languages
+/**
+ * Find emoji flag entry based on country name rather than code
+ * @param {string} countryName
+ * @returns
+ */
+function emojiDataLookup(countryName) {
+  return emojiFlags.data.find((entry) => entry.name === countryName);
+}
+
+/**
+ * This function does a best effort lookup of a locales ISO 3166-1 alpha-2 country code
+ * and returns the corresponding emoji flag.
+ * @param {string} locale the currently assigned locale
+ * @returns {string} the emoji flag
+ */
 function getFlag(locale) {
   if (!locale) return "";
-  const countryCode = locale.split("-")[1];
-  if (!countryCode) return "";
-  return `${emojiFlags.countryCode(countryCode).emoji}  `;
+
+  const countryNames = new Intl.DisplayNames("en", { type: "region" });
+  const [first, second] = locale.split("-").map((part) => part?.toLocaleUpperCase());
+
+  const [firstCountryName, secondCountryName] = [countryNames.of(first ?? "EN"), countryNames.of(second ?? "EN")];
+
+  if (first && firstCountryName !== first) {
+    return `${emojiDataLookup(firstCountryName)?.emoji}  `;
+  } else if (second && secondCountryName !== second) {
+    return `${emojiDataLookup(secondCountryName)?.emoji}  `;
+  } else {
+    switch (locale) {
+      case "en":
+        return `${emojiFlags.countryCode("US")?.emoji}  `;
+      case "nb":
+      case "nn":
+        return `${emojiFlags.countryCode("NO")?.emoji}  `;
+      default:
+        return `🤷  `;
+    }
+  }
 }
 
 export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "" }) {
-  const localeFilePath = join(cwd, "locale", locale) + ".json";
+  const localeFilePath = `${cwd}/locales/${locale}/messages.mjs`;
```

Comment:
we should use path.join to ensure that paths are correct on windows

Reply:
Does the extension have to be `.mjs`, aka does this have to be a common js file? I would prefer this to align with the rest of the application which we advice to be ESM (or TS).

Reply:
> Does the extension have to be `.mjs`, aka does this have to be a common js file? I would prefer this to align with the rest of the application which we advice to be ESM (or TS).

@trygve-lie it's configurable. I'm working on making it possible to extract and compile messages using CLI commands. I'll make sure to add support for different namespaces.